### ISSUE TITLE
Server Listening on TCP IPv6 Does Not Receive Messages

### ISSUE THREAD

**Issue by Kevin-Andrew** [IonxSolutions/serilog-sinks-syslog#13]
If you configure a Syslog server to listen on an IPv6 address using TCP, the `SyslogTcpSink` will not send messages to it via IPv6. For example, when running [Syslog Watcher](https://ezfive.com/syslog-watcher/downloads/) on Windows configured with both IPv4 and IPv6, it can receive messages via the following:
1. IPv4 TCP
2. IPv4 UDP
3. IPv6 UDP

But it will not receive a message when trying to use IPv6 TCP.

Here we see the Windows network configuration of the server that is running the Syslog Watcher:
![image](https://user-images.githubusercontent.com/22459191/105569868-c7e62a80-5d02-11eb-8a87-23eb06876651.png)

Here is the Syslog Watcher configuration:
![image](https://user-images.githubusercontent.com/22459191/105570065-1d6f0700-5d04-11eb-82d8-9d39ed563f79.png)

And here is the client machine from which we'll run the sample application.
![image](https://user-images.githubusercontent.com/22459191/105569897-f8c65f80-5d02-11eb-9722-e21ea9a17c0b.png)

The sample application uses the following code:
```
private void Button1_Click(object sender, EventArgs e)
{
    var config = new SyslogTcpConfig
    {
        // When using TCP with IPv6 it doesn't work.
        Host = "2001:1:1:1::200",

        // When using TCP with IPv4 it does work.
        //Host = "172.21.21.200",
        Formatter = new Rfc5424Formatter(templateFormatter: new Serilog.Formatting.Display.MessageTemplateTextFormatter("TCP: {Message}", null)),
    };

    var log = new LoggerConfiguration()
        .WriteTo.TcpSyslog(config)
        .CreateLogger();

    log.Information("This is a test message.");

    log.Dispose();
}

private void Button2_Click(object sender, EventArgs e)
{
    // When using UDP, both IPv4 and IPv6 work.
    var log = new LoggerConfiguration()
        .WriteTo.UdpSyslog("2001:1:1:1::200")
        .CreateLogger();

    log.Information("This is a test message.");

    log.Dispose();
}
```

Looking at the source code of `SyslogTcpSink`, when using the .NET `TcpClient` (https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient), in order to support both IPv4 and IPv6, we must specify `AddressFamily.InterNetworkV6` in the constructor and set the `Client.DualMode` property to `true` (https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.dualmode?view=netstandard-1.6). A blog about this can be found at: http://www.lybecker.com/blog/2018/08/23/supporting-ipv4-and-ipv6-dual-mode-network-with-one-socket/

I have done a test after modifying the source code of the `SyslogTcpSink` class:
```
this.client = new TcpClient(AddressFamily.InterNetworkV6);
this.client.Client.DualMode = true;
```
and then the server will receive a message via TCP on the IPv6 address. I have only tested this on Windows.

**Comment by colin-anderson** [comment]
Thanks for opening this issue, and I really appreciate you digging in too! 🙏 

I'll try to find some time in the next few days, and try to reproduce and fix - will update once that's done.

**Comment by Kevin-Andrew** [comment]
Cool, thanks. Feel free to reach out if you need anything else, testing or otherwise.

**Comment by colin-anderson** [comment]
Resolved via #16
